sdk/core/azure-core/azure/core/pipeline/__init__.py

```diff
@@ -50,7 +50,17 @@ def __exit__(self, exc_type, exc_value, traceback):
 
 
 class PipelineContext(dict):
+    """A context object carried by the pipeline request and
+     response containers.
 
+    This is transport specific and can contain data persisted between
+    pipeline requests (for example reusing an open connection pool or "session"),
+    as well as used by the SDK developer to carry arbitrary data through
+    the pipeline.
+
+    :param transport: The HTTP transport type.
+    :param dict options: Developer-defined keyword arguments.
```

Comment:
This param seems confusing. There is not a single `options` param, rather any of the (zero or more) additional keyword arguments comprise what becomes the options dict